vng_api_common/authorizations/serializers.py

```diff
@@ -0,0 +1,94 @@
+import logging
+
+from django.db import transaction
+from django.utils.translation import ugettext_lazy as _
+
+from rest_framework import serializers
+
+from vng_api_common.authorizations.models import Applicatie, Autorisatie
+
+logger = logging.getLogger(__name__)
+
+
+class AutorisatieSerializer(serializers.HyperlinkedModelSerializer):
+    class Meta:
+        model = Autorisatie
+        fields = (
+            'component',
+            'zaaktype',
+            'scopes',
+            'max_vertrouwelijkheidaanduiding',
+        )
+
+
+class ApplicatieSerializer(serializers.HyperlinkedModelSerializer):
+    autorisaties = AutorisatieSerializer(many=True, required=False)
+
+    class Meta:
+        model = Applicatie
+        fields = (
+            'url',
+            'client_ids',
+            'label',
+            'heeft_alle_autorisaties',
+            'autorisaties'
+        )
+        extra_kwargs = {
+            'url': {
+                'lookup_field': 'uuid',
+            },
+            'heeft_alle_autorisaties': {
+                'required': False,
+            }
+
+        }
+
+    def validate(self, attrs):
+        validated_attrs = super().validate(attrs)
+
+        # either autorisaties or heeft_alle_autorisaties can be specified
+        autorisaties_obj = None
+        heeft_alle_autorisaties_obj = None
+        # in case of update:
+        if self.instance:
+            autorisaties_obj = self.instance.autorisaties.all()
+            heeft_alle_autorisaties_obj = self.instance.heeft_alle_autorisaties
+
+        autorisaties = validated_attrs.get('autorisaties', None) or autorisaties_obj
+        heeft_alle_autorisaties = validated_attrs.get('heeft_alle_autorisaties', None) or heeft_alle_autorisaties_obj
+
+        if autorisaties and heeft_alle_autorisaties is True:
+            raise serializers.ValidationError(
+                    {'nonFieldErrors': _('Either autorisaties or heeft_alle_autorisaties can be specified')},
```

Comment:
I think you can raise the exception directly as a string instead of a dict, and it'll put it automatically in the `nonFieldErrors`.

We might have to change the key `nonFieldErrors` into something different at some point because of country-wide API-strategy, and I'd like to have that controlled in a single location :-)

Reply:
dict is replaced with str